site/profile/manifests/android.pp

```diff
@@ -0,0 +1,33 @@
+
+class profile::android (
+        String $sdk_tools,
+        String $sdk_path,
+	Array[String] $npm_packages,
+	Array[String] $sdk_packages,
+)
+
+{
+  archive { "/tmp/$sdk_tools":
+    source       => "https://dl.google.com/android/repository/$sdk_tools",
+    extract      => true,
+    extract_path => $sdk_path,
+    user          => 'ubuntu',
+    group         => 'ubuntu',
+    }
+  $npm_packages.each |String $npm_package| {
+    exec { "install_global_npm_modules_${npm_package}":
+      command  => "sudo npm i -g \"${npm_package}\"",
```

Comment:
You should use a normal package resource with the npm provider (there's a module for it).
Or, at least, use creates, unless or onlyif parameters to avoid exec command to be run every time.